### ISSUE TITLE
代码更新问题

### ISSUE THREAD

**Issue by AlvinAi96** [AlvinAi96/match_rcnn#3]
我们团队又回顾了下train_mm.py的代码，发现有几个问题：

**Comment by AlvinAi96** [comment]
``请自行更新网络结构
class MatchModel(nn.Module):
    def __init__(self):
        super(MatchModel, self).__init__()
        self.conv1 = nn.Conv2d(256, 256, 3, padding=1)
        self.conv2 = nn.Conv2d(256, 256, 3, padding=1)
        self.conv3 = nn.Conv2d(256, 256, 3, padding=1)
        self.conv4 = nn.Conv2d(256, 1024, 3, padding=1)
        self.relu1 = nn.ReLU(inplace=True)
        self.relu2 = nn.ReLU(inplace=True)
        self.relu3 = nn.ReLU(inplace=True)
        self.relu4 = nn.ReLU(inplace=True)
        # 就是在卷积层之后，用GAP替代FC全连接层。原因有两个点：
        # 一是GAP在特征图与最终的分类间转换更加简单自然；
        # 二是不像FC层需要大量训练调优的参数，降低了空间参数会使模型更加健壮，抗过拟合效果更佳。
        self.globalavgpool = nn.AdaptiveAvgPool2d((1,1)) # nx1024x7x7 -> nx1024x1x1
        self.relu5 = nn.ReLU(inplace=True)
        self.fc1 = nn.Linear(1024, 256)
        self.fc2 = nn.Linear(256, 1)
        self.sigmoid = nn.Sigmoid()      
    def init_weights(self):
        for m in self.modules():
            if isinstance(m, nn.Conv2d):
                nn_init.xavier_uniform_(m.weight)
                nn.init.constant_(m.bias, 0)
            elif m == self.fc2:
                nn.init.normal_(m.weight, 0, 0.01)
                nn.init.constant_(m.bias, 0) 
            else:
                nn_init.xavier_uniform_(m.weight)
                nn.init.constant_(m.bias, 0)            
    def forward(self, x1, x2):
        # feature preparation part for anchor feature x1
        x1 = self.conv1(x1)
        x1 = self.relu1(x1)
        x1 = self.conv2(x1)
        x1 = self.relu2(x1)
        x1 = self.conv3(x1)
        x1 = self.relu3(x1)
        x1 = self.conv4(x1)
        x1 = self.relu4(x1)
        x1 = self.globalavgpool(x1)
        x1 = self.relu5(x1)
        x1 = x1.view(x1.size(0),-1) # [n, 1024]
        x1 = self.fc1(x1) # [n,256]
        print(x1.size())
        # feature preparation part for pair feature x2
        x2 = self.conv1(x2)
        x2 = self.relu1(x2)
        x2 = self.conv2(x2)
        x2 = self.relu2(x2)
        x2 = self.conv3(x2)
        x2 = self.relu3(x2)
        x2 = self.conv4(x2)
        x2 = self.relu4(x2)
        x2 = self.globalavgpool(x2)
        x2 = self.relu5(x2)
        x2 = x2.view(x2.size(0),-1) # [n, 1024]
        x2 = self.fc1(x2) # [n, 256]
        print(x2.size())
        # similarity learning net
        x = torch.sub(x1, x2) # Subtraction
        x = torch.mul(x, x) # dot mulitply - Element-wise square
        x = self.fc2(x)
        x = self.sigmoid(x)
        return x
